src/core/autosave/autosave.Hook.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useRef, useState } from 'react';
+
+import {
+  useCanvasSchemaContext,
+  useCanvasViewSettingsContext,
+} from '@/core/providers';
+
+import { saveToLocal, retrieveFromLocal } from './autosave.business';
+
+const useAutosave = () => {
+  const { canvasSchema, setCanvasSchema } = useCanvasSchemaContext();
+  const { filename, setLoadSample } = useCanvasViewSettingsContext();
+
+  const INTERVAL = 6000;
+  const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+  const [isSaving, setIsSaving] = useState<boolean>(false);
```

Comment:
Mmm... not sure if we need _isSaving_ on the one hand javaScript is singlethreaded by default, in the other save to local storage not sure if it's blocking or not (I/O)